aioseop_class.php

```diff
@@ -4301,153 +4305,134 @@ function apply_description_format( $description, $post = null ) {
 		}
 
 		if (
-				(
-					is_category() &&
-					! empty( $aioseop_options['aiosp_category_noindex'] )
-				) ||
-				(
-					! is_category() &&
-					is_archive() &&
-					! is_tag() &&
-					! is_tax() &&
-					(
-						(
-							is_date() &&
-							! empty( $aioseop_options['aiosp_archive_date_noindex'] )
-						) ||
-						(
-							is_author() &&
-							! empty( $aioseop_options['aiosp_archive_author_noindex'] )
-						)
-					)
-				) ||
-				(
-					is_tag() &&
-					! empty( $aioseop_options['aiosp_tags_noindex'] )
-				) ||
-				(
-					is_search() &&
-					! empty( $aioseop_options['aiosp_search_noindex'] )
-				) ||
-				(
-					is_404() &&
-					! empty( $aioseop_options['aiosp_404_noindex'] )
-				) ||
-				(
-					is_tax() &&
-					in_array( get_query_var( 'taxonomy' ), $tax_noindex )
-				)
+				( is_category() && ! empty( $aioseop_options['aiosp_category_noindex'] ) ) ||
+				( is_date() && ! empty( $aioseop_options['aiosp_archive_date_noindex'] ) ) ||
+				( is_author() && ! empty( $aioseop_options['aiosp_archive_author_noindex'] ) ) ||
+				( is_tag() && ! empty( $aioseop_options['aiosp_tags_noindex'] ) ) ||
+				( is_search() && ! empty( $aioseop_options['aiosp_search_noindex'] ) ) ||
+				( is_404() && ! empty( $aioseop_options['aiosp_404_noindex'] ) ) ||
+				( is_tax() && in_array( get_query_var( 'taxonomy' ), $tax_noindex ) )
 		) {
-			$noindex = 'noindex';
+			$noindex = true;
+		}
 
-			// #322: duplicating this code so that we don't step on some other entities' toes.
-			if (
-					( 'on' === $aiosp_nofollow ) ||
-					(
-						( ! empty( $aioseop_options['aiosp_paginated_nofollow'] ) ) &&
-						$page > 1
-					) ||
-					(
-							( '' === $aiosp_nofollow ) &&
-							( ! empty( $aioseop_options['aiosp_cpostnofollow'] ) ) &&
-							in_array( $post_type, $aioseop_options['aiosp_cpostnofollow'] )
-					)
-			) {
-				$nofollow = 'nofollow';
-			}
-			// #322: duplicating this code so that we don't step on some other entities' toes.
-		} elseif (
+		if (
 				is_single() ||
 				is_page() ||
 				$this->is_static_posts_page() ||
 				is_attachment() ||
-				is_category() ||
-				is_tag() ||
-				is_tax() ||
-				( $page > 1 ) ||
 				$this->check_singular()
 		) {
-			$post_type = get_post_type();
-			if (
-					$aiosp_noindex ||
-					$aiosp_nofollow ||
-					! empty( $aioseop_options['aiosp_cpostnoindex'] ) ||
-					! empty( $aioseop_options['aiosp_cpostnofollow'] ) ||
-					! empty( $aioseop_options['aiosp_paginated_noindex'] ) ||
-					! empty( $aioseop_options['aiosp_paginated_nofollow'] )
-			) {
-
-				if (
-						( 'on' === $aiosp_noindex ) ||
-						(
-							( ! empty( $aioseop_options['aiosp_paginated_noindex'] ) )
-							&& $page > 1
-						) ||
-						(
-							( '' === $aiosp_noindex ) &&
-							( ! empty( $aioseop_options['aiosp_cpostnoindex'] ) ) &&
-							in_array( $post_type, $aioseop_options['aiosp_cpostnoindex'] )
-						)
-
+			if ( ! $noindex ) {
```

Comment:
I believe this should be `if ( $noindex ) {`.

Reply:
It should be `! noindex`.

On lines 4346-4351, the code first checks whether the page has been noindexed/nofollowed via our meta box on the Edit screen; so it checks the postmeta values. If it has been, then we don't need to run subsequent checks for either noindex/nofollow (or both - see lines 4361-4364). 

Only if the page hasn't been noindexed/nofollowed individually should we check whether its type has been noindexed/nofollowed globally/site-wide in the General Settings menu.